package.json

```diff
@@ -4,40 +4,50 @@
   "description": "The standard ROS Javascript Library",
   "version": "1.1.0",
   "license": "BSD-2-Clause",
+  "type": "module",
   "main": "./src/RosLibNode.js",
+  "// browser comment": "rollup uses this for shimming so that the browser bundle uses the global builtins",
   "browser": {
     "./src/RosLibNode.js": "./src/RosLib.js",
+    "socket.io": "./src/util/shim/socket.io.js",
     "canvas": "./src/util/shim/canvas.js",
     "ws": "./src/util/shim/WebSocket.js",
     "xmldom": "./src/util/shim/xmldom.js",
     "./src/util/decompressPng.js": "./src/util/shim/decompressPng.js"
   },
   "devDependencies": {
+    "@babel/eslint-parser": "^7.15.4",
+    "@rollup/plugin-commonjs": "^20.0.0",
+    "@rollup/plugin-json": "^4.1.0",
+    "@rollup/plugin-node-resolve": "^13.0.4",
     "chai": "*",
+    "eslint": "^8.0.0-beta.2",
     "grunt": "^1.0.0",
-    "grunt-browserify": "^6.0.0",
     "grunt-cli": "^1.2.0",
     "grunt-contrib-clean": "^2.0.0",
-    "grunt-contrib-jshint": "^3.0.0",
     "grunt-contrib-uglify": "^5.0.0",
     "grunt-contrib-watch": "^1.0.0",
     "grunt-jsdoc": "^2.1.0",
     "grunt-karma": "^4.0.0",
     "grunt-mocha-test": "^0.13.0",
+    "grunt-shell": "^3.0.1",
     "karma": "^6.3.0",
     "karma-chai": "^0.1.0",
     "karma-firefox-launcher": "^2.1.0",
     "karma-mocha": "^2.0.0",
     "load-grunt-tasks": "^5.1.0",
     "mocha": "^9.0.1",
+    "rollup": "^2.56.3",
+    "rollup-plugin-filesize": "^9.1.1",
+    "rollup-plugin-terser": "^7.0.2",
     "time-grunt": "^2.0.0"
   },
   "dependencies": {
     "cbor-js": "^0.1.0",
     "eventemitter2": "^6.4.0",
     "object-assign": "^4.0.0",
     "pngparse": "^2.0.0",
-    "socket.io": "^4.0.0",
+    "socket.io": "trusktr/socket.io#bf4bd150cff3c043c1f60ddf347e1b3ef0f1339a",
```

Comment:
This is not idea (see https://github.com/socketio/socket.io/pull/4100). The alternative would be to update the `socket.io` import sites to import from a local file that shims the missing default function (`(srv, opts) => new Server(srv, opts);`) in Node.js, but in the browser just uses the `io` global.